### PR TITLE
update bug reporting and contributing info

### PR DESCRIPTION
**Pull request by raffenet**

### CODE REVIEW

File: CONTRIBUTING [hunk 1]
```diff
@@ -0,0 +1,9 @@
+How to contribute to MPICH
+
+1. Sign and subnit a contributors' agreement. http://www.mpich.org/documentation/contributor-docs/
```
Comment (wesbland) [inline CONTRIBUTING]: submit

Comment (wesbland) [reply CONTRIBUTING]: You might want to consider something like this: https://cla-assistant.io

I assume this is similar to what libfabric uses (if not the same thing).

File: CONTRIBUTING [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+2. Format your patches according to the MPICH coding standards. https://wiki.mpich.org/mpich/index.php/Coding_Standards
```
Comment (wesbland) [inline CONTRIBUTING]: Those standards might not be up to date. I'd mention that we have a script to enforce these things.

File: CONTRIBUTING [hunk 3]
```diff
@@ -0,0 +1,11 @@
+How to contribute to MPICH
+
+1. Sign and submit a contributors' agreement. http://www.mpich.org/documentation/contributor-docs/
+
+2. Format your patches according to the MPICH coding standards. https://wiki.mpich.org/mpich/index.php/Coding_Standards
+
+3. Install this pre-commit hook (https://gist.github.com/wesbland/3546c4483ceffe129a6e7fc4105fbf75) in your local git repository to catch whitespace violations in new code. Patches with improper whitespace may be automatically rejected for inclusion in MPICH.
```
Comment (pavanbalaji) [inline CONTRIBUTING]: Can we put a copy of it on the `mpich` github (with credit to Wesley's github)?

Comment (wesbland) [reply CONTRIBUTING]: Yeah, I was kinda thinking the same thing. While I love the glory that comes with creating the hook 🙄, I'd love to hand off the maintenance.

Comment (raffenet) [reply CONTRIBUTING]: Just coming back to this. Will do.

Comment (raffenet) [reply CONTRIBUTING]: Oh wait, we put the hook into the git repo. I should update the instructions.

**Comment by raffenet** [comment]
Updated the `CONTRIBUTING` file now that the pre-commit hook is included with MPICH.

**Pull request merged**